### ISSUE TITLE
Idea for XUID fix

### ISSUE THREAD

**Issue by thomcrielaard** [GeyserMC/Floodgate#357]
Just an idea; so don't flame me please :). 

What if you were to add a temporary whitelist for the fwhitelist command if the user has never played on the server. Whenever someone adds their username to the fwhitelist you store the username temporarily. Then whenever Geyser performs a whitelist check, and the username of the connecting (Bedrock) user matches the name in your temporary whitelist file, you add that player to the whitelist

**Comment by jackson-57** [comment]
Good idea! I've wondered about the same too. I'd suggest adding an expiration for temporary whitelist entries to limit the potential for abuse if a temporary entry is forgotten about, for example.

**Comment by thomcrielaard** [comment]
Good suggestion! Totally agree with you!

**Comment by Konicai** [comment]
I'll give this a shot.

**Comment by Tim203** [comment]
An issue like this one has been open for much longer than I wanted and expected, and it's as we speak still not fixed.
However, between the time that this issue was created and now we've seen other APIs that can properly use the Xbox API.
So we'd like to fix it properly instead of working around the issue, even though the idea itself is quite good and would definitely work.
The global api (our rest api) is currently being rewritten. And we're going to try it again with that rewrite, before the release of Floodgate & Geyser 3.0.
This however means that I'm going to close this issue specifically, and I'll probably post the updates to #304 instead.
